### PR TITLE
Updates and Fixes for Windows Experience [API-1455]

### PR DESCRIPTION
**Pull request by utku-caglayan**
1. Makes few fixes for bugs encountered in Windows testing
2 .Updates release pipeline to generate a Windows executable with metadata information

Output of the mentioned pipeline can be seen here:
https://github.com/utku-caglayan/hazelcast-commandline-client/releases/tag/v1.0.0-beta3

### CODE REVIEW

File: winres/winres.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    "OTHER": {
+      "0000": "hazelcast-icon-flat-sm.png"
+    }
+  },
+  "RT_MANIFEST": {
+    "#1": {
+      "0409": {
+        "identity": {
+          "name": "",
+          "version": ""
```
Comment (yuce) [inline winres/winres.json:12]: Do we need to set these?

File: winres/winres.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        },
+        "description": "",
+        "minimum-os": "win7",
+        "execution-level": "as invoker",
```
Comment (yuce) [inline winres/winres.json:16]: What does `as invoker` do?

File: winres/winres.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        "use-common-controls-v6": false
+      }
+    }
+  },
+  "RT_VERSION": {
+    "#1": {
+      "0409": {
+        "info": {
+          "0409": {
+            "FileDescription": "Command line client for Hazelcast",
```
Comment (yuce) [inline winres/winres.json:37]: How about "Hazelcast Command Line Client" ?

File: winres/winres.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            "LegalCopyright": "©Hazelcast, Inc.",
+            "OriginalFilename": "hzc.exe",
+            "ProductName": "hzc",
+            "Comments": "",
+            "CompanyName": "Hazelcast, Inc.",
+            "FileVersion": "",
+            "LegalTrademarks": "",
+            "PrivateBuild": "",
+            "ProductVersion": "",
+            "SpecialBuild": ""
```
Comment (yuce) [inline winres/winres.json:48]: What's the difference between `FileVersion` and `ProductVersion`? Should we put the CLC version in one/both of them?

File: types/mapcmd/put-all.go [hunk 5]
```diff
@@ -60,8 +60,8 @@ func NewPutAll(config *hazelcast.Config) *cobra.Command {
 		if len(mapKeys) != 0 ||
 			len(mapValues) != 0 ||
 			len(mapValueFiles) != 0 ||
-			mapKeyType != "" ||
-			mapValueType != "" {
+			mapKeyType != "string" ||
+			mapValueType != "string" {
 			return hzcerrors.NewLoggableError(nil, fmt.Sprintf("%s is already set, there cannot be additional flags", JSONEntryFlag))
```
Comment (yuce) [inline types/mapcmd/put-all.go:65]: Why is this change required? Does it make sense to move this validation to its own function?

File: winres/winres.json (hunk 1 above)
Comment (utku-caglayan) [reply winres/winres.json:12]: These are the default values created when you run "go-winres init". I left them unset so that we can just see the fields if we want to change anything

File: winres/winres.json (hunk 2 above)
Comment (utku-caglayan) [reply winres/winres.json:16]: I'm not sure about the possible values but I think this is where you specify if this executable is intended to work like a service or invocation base

File: internal/go-prompt/render.go [hunk 6]
```diff
@@ -82,6 +82,11 @@ func (r *Render) prepareArea(lines int) {
 func (r *Render) UpdateWinSize(ws *WinSize) {
 	r.row = ws.Row
 	r.col = ws.Col
+	// don't extend the prompt all the way to the right.
+	// this eliminates problems on terminals with big fonts
+	if r.col > 3 {
+		r.col -= 3
+	}
```
[the remark this answers is not in the document]
Comment (utku-caglayan) [reply internal/go-prompt/render.go:89]: Yes I thought that would eliminate the problem we are facing on Windows. Reverting this

File: winres/winres.json (hunk 3 above)
Comment (utku-caglayan) [reply winres/winres.json]: Thanks see https://github.com/hazelcast/hazelcast-commandline-client/pull/118/commits/a5e0fcc2528d594f911eada5befecc6539550643

File: winres/winres.json [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    }
+  },
+  "RT_VERSION": {
+    "#1": {
+      "0409": {
+        "info": {
+          "0409": {
+            "FileDescription": "Command line client for Hazelcast",
+            "InternalName": "hzc",
+            "LegalCopyright": "©Hazelcast, Inc.",
```
[the remark this answers is not in the document]
Comment (utku-caglayan) [reply winres/winres.json]: Thanks see https://github.com/hazelcast/hazelcast-commandline-client/pull/118/commits/a5e0fcc2528d594f911eada5befecc6539550643

File: winres/winres.json [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+            "OriginalFilename": "hzc.exe",
+            "ProductName": "hzc",
```
[the remark this answers is not in the document]
Comment (utku-caglayan) [reply winres/winres.json]: Thanks see https://github.com/hazelcast/hazelcast-commandline-client/pull/118/commits/a5e0fcc2528d594f911eada5befecc6539550643

File: winres/winres.json (hunk 4 above)
Comment (utku-caglayan) [reply winres/winres.json]: I removed them since we pass them via command flag to "go-winres"

Comment (utku-caglayan) [reply winres/winres.json]: https://github.com/hazelcast/hazelcast-commandline-client/pull/118/commits/a5e0fcc2528d594f911eada5befecc6539550643

File: internal/go-prompt/render.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (utku-caglayan) [reply internal/go-prompt/render.go]: Reverted https://github.com/hazelcast/hazelcast-commandline-client/pull/118/commits/a5e0fcc2528d594f911eada5befecc6539550643

File: types/mapcmd/put-all.go (hunk 5 above)
Comment (utku-caglayan) [reply types/mapcmd/put-all.go]: Tried to refactor for clarity. But I did not understand how to move this to a function already in a function. Please let me know if I can make it better.

Comment (yuce) [reply types/mapcmd/put-all.go]: OK, I didn't realize it was already in a function.

File: winres/winres.json [hunk 9]
```diff
@@ -0,0 +1,52 @@
+{
+  "RT_GROUP_ICON": {
+    "OTHER": {
+      "0000": "hazelcast-icon-flat-sm.png"
+    }
+  },
+  "RT_MANIFEST": {
+    "#1": {
+      "0409": {
+        "identity": {
```
Comment (yuce) [inline winres/winres.json:10]: In the docs it says:

> It is recommended to omit identity if your program is a plain application, not meant to be a side-by-side dependency.

So maybe we can leave that out?

File: winres/winres.json (hunk 2 above)
Comment (yuce) [reply winres/winres.json:16]: It seems `execution-level` can have three values: https://github.com/tc-hib/winres/blob/e87828d02c266dfb1edac897bf51090a9ae44971/manifest.go#L80

File: winres/winres.json (hunk 9 above)
Comment (utku-caglayan) [reply winres/winres.json]: https://github.com/hazelcast/hazelcast-commandline-client/pull/118/commits/3133e5e22d274ecc42ad27854d72998c1985632e

File: types/mapcmd/map.go [hunk 10]
```diff
@@ -241,7 +241,7 @@ func decorateCommandWithMapNameFlags(cmd *cobra.Command, mapName *string, requir
 }
 
 func decorateCommandWithMapKeyTypeFlags(cmd *cobra.Command, mapKeyType *string, required bool) {
-	cmd.Flags().StringVar(mapKeyType, MapKeyTypeFlag, "string", fmt.Sprintf("type of the key, one of: %s", strings.Join(internal.SupportedTypeNames, ",")))
+	cmd.Flags().StringVar(mapKeyType, MapKeyTypeFlag, "", fmt.Sprintf("type of the key [default string], one of: %s", strings.Join(internal.SupportedTypeNames, ",")))
```
[the remark this answers is not in the document]
Comment (utku-caglayan) [reply types/mapcmd/map.go]: Thanks, here https://github.com/hazelcast/hazelcast-commandline-client/pull/118/commits/3133e5e22d274ecc42ad27854d72998c1985632e

**Review by yuce — APPROVED** [review]
Looks good!
